File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel_#{params['room']}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    Message.create! 'room_channel', message: data['message'], user_id: current_user.id, room_id: params['room']
  end

end

File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

// $(function()
document.addEventListener('turbolinks: load', () => {
  consumer.subscriptions.create({channel: "RoomChannel", room: $('#messages').data('room_id') }, {
    connected() {
      // Called when the subscription is ready for use on the server
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received: function(data) {
      return $('#messages').append(data['message']);
    },

    speak: function(message) {
      return this.perform('speak', {message: message});
    }
  });
})